Example/Backpack/UIKit/Components/Card/Card Button/CardButtonsViewController.swift

```diff
@@ -0,0 +1,280 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import UIKit
+import Backpack
+
+private struct CardButtonSampleRow {
+    let title: String
+    let cardButtons: [CardButtonSample]
+}
+
+private struct CardButtonSample {
+    enum Icon {
+        case share
+        case unsaved
+        case saved
+    }
+
+    let icon: Icon
+    let backgroundColor: UIColor
+    let size: BPKCardButtonSize
+    let style: BPKCardButtonStyle
+}
+
+final class CardButtonsViewController: UIViewController {
+    private let sampleButtons: [CardButtonSampleRow] = [
+        CardButtonSampleRow(
```

Comment:
thought:
I think this may be a bit over generalised, 2 small private static methods to make the save or share button instances may have sufficed and make the row creation process less verbose, please consider it

Reply:
maybe even one method/computed-var for each different type, with this, from the caling side is clear what your're creating, and in the implementation side, you avoid having logic for different enum cases and so on.
```swift
let buttons = [.onDark(.share), .contained(.save), .contained(.saveSmall)]

private extension CardButtonSampleRow {
  static func onDark(CardButtonSample) { ... }
  static func contained(CardButtonSample) { ... }
}
private extension CardButtonSample {
   static var shareSmall: CardButtonSample { ... }
   static var save: CardButtonSample { ... }
   static var saveChecked: CardButtonSample { ... }
}
```

wdyt?

Reply:
This is the change 13fd5bfed2e7093e80ce6e150a0d38f827068c68